e2e/advanced/environment_test.go

```diff
@@ -39,7 +40,58 @@ import (
 	"github.com/apache/camel-k/v2/pkg/util/defaults"
 )
 
-func TestEnvironmentTrait(t *testing.T) {
+func TestEnvironmentTraitVars(t *testing.T) {
+	t.Parallel()
+
+	WithNewTestNamespace(t, func(ctx context.Context, g *WithT, ns string) {
+		operatorID := "camel-k-trait-environment"
+		g.Expect(CopyCamelCatalog(t, ctx, ns, operatorID)).To(Succeed())
+		g.Expect(CopyIntegrationKits(t, ctx, ns, operatorID)).To(Succeed())
+		g.Expect(KamelInstallWithID(t, ctx, operatorID, ns)).To(Succeed())
+
+		g.Eventually(SelectedPlatformPhase(t, ctx, ns, operatorID), TestTimeoutMedium).Should(Equal(v1.IntegrationPlatformPhaseReady))
+
+		// Create configmap
+		var cmData = make(map[string]string)
+		cmData["my-cm-key"] = "hello configmap"
+		err := CreatePlainTextConfigmap(t, ctx, ns, "my-cm", cmData)
+		require.NoError(t, err)
```

Comment:
This is more for a unit test. I think we better use the gomega funcs to assert no errors.

Reply:
done